### DISCUSSION TITLE
Pre-dependent active trigger(前置依赖主动触发后续工作流)

### DISCUSSION THREAD

**Discussion by mpmiles** [apache/dolphinscheduler#6386] [Q&A]
Whether the DS supports multiple workflows that are dependent in advance before triggering subsequent workflows after they are completed, rather than through timed tasks. If not, which God to help provide an idea, thank you

---
DS是否支持前置依赖的多个工作流结束后再触发后续工作流，而不是通过定时任务。如不支持，哪位大神帮忙提供个思路，多谢

**Comment by davidzollo** [comment]
Did you try sub-workflow?

**Reply by mpmiles** [reply]
Sub-workflow can trigger dependencies within a single workflow, but if the dependencies between workflows are complex, multiple workflows must be completed before the post-workflow can be triggered, or a workflow completed after the need to trigger post-dependent workflow, can not be placed in a sub-workflow repeated execution of it? What needs to be considered is when each task completes, scans its dependency actively, the downstream satisfies the dependency, triggers automatically

**Comment by sakinsun** [comment]
不是有一个 DEPENDENT 节点么？里面就可以把你的前置依赖设置好，只有前置依赖满足了才会跑后面的节点

**Reply by mpmiles** [reply]
首先非常感谢你的回复。但是
我的问题是要主动触发后置依赖的工作流，使用depend节点在同一个工作流里是可以触发后边的依赖，但是也是需要将此工作流设置成一个定时任务调起，然后去检查前置depend任务，而不是说，前置都结束后自己就触发起来后置。举个例子：A依赖B、C工作流，B和C都都结束了，不需要定时，不需要人为干预，直接就可以触发A。

**Comment by zichen-fei** [comment]
https://github.com/scxwhite/hera
可以参考下hera的任务分发
